File: Logic_Building_Assignments/Logic_Building_Assignment_5/Assignment5_4.c
//  4. Write a program which accepts N from user and print all odd numbers up to N.

#include<stdio.h>

void OddDisplay(int iNo)
{
    int i = 0;

    for(i = 1; i <= iNo; i++)
    {
        if((i % 2) != 0)
        {
            printf("%d\t",i);
        }
    }
}

int main()
{
    int iValue = 0;

    printf("Enter a number : ");
    scanf("%d",&iValue);

    OddDisplay(iValue);

    return 0;
}

/*

    Output:
    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 5>gcc Assignment5_4.c 

    D:\Marvellous Infosystems\Logic Building with Industrial Project Batch\Logic Building Assignments\Logic Building Assignment 5>a
    Enter a number : 18
    1       3       5       7       9       11      13      15      17

*/

File: Python_Assignments/Python_Assignment_5/Assignment5_4.py
#   4. Write a recursive program which accept number from user and return
#      summation of its digits.

sum = 0

def DigitSum(No):
    global sum
    if No != 0:
        rem = No%10
        sum = sum + rem
        DigitSum(int(No/10))
    return sum

def main():
    Number = int(input("Enter your number : "))
    Result = DigitSum(Number)
    print("Summation of digits : ",Result)

if __name__ == "__main__":
    main()

#   Output:
#   D:\Python\Marvellous Python\Python Assignments\Python Assignment 5>Python Assignment5_4.py
#   Enter your number : 879
#   Summation of digits :  24